course5/homework12/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+FIND_PACKAGE(Boost COMPONENTS program_options REQUIRED)
+INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIRS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -lboost_program_options")
```

Comment:
you don't add libraries by setting compilation flags, but rather by using ```target_link_libraries```.
Setting explicit compilation flags should be avoided, as this breaks portability (GCC vs Clang is usually OK, but think of Visual Studio!).